### PR TITLE
Updating 3.0 breaking changes doc

### PR DESCRIPTION
**Pull request by ajcvickers**
Including content from @AndriySvyryd and @bricelam 

@divega Can you review?

### CODE REVIEW

File: entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md [hunk 1]
```diff
@@ -11,29 +11,106 @@ uid: core/what-is-new/ef-core-3.0/breaking-changes
...
+```C#
+context.ChangeTracker.CascadeDeleteTiming = CascadeTiming.OnSaveChanges;
+context.ChangeTracker.DeleteOrphansTiming = CascadeTiming.OnSaveChanges;
+```
+
+### Query types and entity types are consolidated into `IEntityType`
+
+[Tracking Issue #14194](https://github.com/aspnet/EntityFrameworkCore/issues/14194)
+
+[Query types](xref:core/modeling/query-types) were introduced in 2.1 as a means to query data that doesn't contain a primary key in a structured way. However it caused some confusion when deciding whether to configure something as an entity type or a query type. Therefor we have decided to bring the two closer together in the API.
```
Comment (divega) [inline entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md]: Therefore

File: entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md [hunk 2]
```diff
@@ -11,29 +11,106 @@ uid: core/what-is-new/ef-core-3.0/breaking-changes
...
 > Please note that the feature sets and schedules of future releases are always subject to change, and although we will try to keep this page up to date, it may not reflect our latest plans at all times.
 
-## SQLite
+These are breaks in either API or behavior between the 2.2.x releases and the 3.0.0 release. Breaks in new features introduced from one 3.0 preview to another 3.0 preview are not documented here.
 
-* Microsoft.EntityFrameworkCore.Sqlite now depends on SQLitePCLRaw.bundle_e_sqlite3 instead of SQLitePCLRaw.bundle_green. This makes the version of SQLite used on iOS consistent with other platforms.
-* Removed SqliteDbContextOptionsBuilder.SuppressForeignKeyEnforcement(). EF Core no longer sends `PRAGMA foreign_keys = 1` when a connection is opened. Foreign keys are enabled by default in SQLitePCLRaw.bundle_e_sqlite3. If you're not using that, you can enable foreign keys by specifying `Foreign Keys=True` in your connection string.
+## High-impact breaks
+
+TODO: Removal of automatic client-evaluation of queries
```
Comment (divega) [inline entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md]: @ajcvickers @smitpatel what do you think we should have here? I think we should have something saying that the change isn't fully implemented yet but then explaining what it is.

Comment (AndriySvyryd) [reply entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md]: I don't think we need anything in the docs until we ship it

Comment (ajcvickers) [reply entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md]: Here's a crazy idea: switch off automatic client evaluation even before query 3.0 is ready...

Comment (divega) [reply entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md]: @ajcvickers I think that may be a good crazy idea. Let’s discuss it today.

File: entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md [hunk 3]
```diff
@@ -107,10 +188,170 @@ public class Order
...
+
+[Tracking Issue #12366](https://github.com/aspnet/EntityFrameworkCore/issues/12366)
 
 `ForSqlServerHasIndex().ForSqlServerInclude()` provided a way to configure columns used with `INCLUDE`, now the same can be accomplished with `HasIndex().ForSqlServerInclude()`
+
+## Breaks that should only impact database providers
+
+Changes that we expect to only impact database providers are documented under [provider changes](../../providers/provider-log.md).
+
+## Breaks to the `Microsoft.Data.Sqlite` ADO.NET database provider
```
[the remark this answers is not in the document]
Comment (ajcvickers) [reply entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md]: @bricelam Sorry--I should have read what I was editing. :-) I'll fix it up later today.

I think we should have a separate page for the ADO.NET changes. @divega?

File: entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md (hunk 2 above)
Comment (roji) [reply entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md]: FWIW I think it's a great idea.

File: entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (divega) [reply entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md]: In theory yes, but I don't know where it belongs...

**Pull request merged**